### PR TITLE
Fix an issue where smspsec might dereference null

### PR DESCRIPTION
**Pull request by eivindjahren**

### CODE REVIEW

File: lib/resdata/rd_smspec.cpp [hunk 1]
```diff
@@ -856,15 +856,18 @@ static void rd_smspec_load_restart(rd_smspec_type *rd_smspec,
...
-        for (i = 0; i < rd_kw_get_size(restart_kw); i++)
-            strcat(tmp_base, (const char *)rd_kw_iget_ptr(restart_kw, i));
-
-        restart_base = util_alloc_strip_copy(tmp_base);
+        int num_blocks = rd_kw_get_size(restart_kw);
+        char *tmp_base = util_calloc(8 * num_blocks + 1, sizeof *tmp_base);
+        for (int i = 0; i < num_blocks; i++) {
+            const char* part = (const char *)rd_kw_iget_ptr(restart_kw, i));
+            if (strlen(part) != 8) {
+                util_abort("%s: unexpcted size of RESTART block:%s\n",
```
Comment (berland) [inline lib/resdata/rd_smspec.cpp:864]: typo unexpected

**Review by berland — APPROVED** [review]
🐛

**Comment by berland** [comment]
amend commit message to explain the consequences of this fix

**Comment by eivindjahren** [comment]
> amend commit message to explain the consequences of this fix

Done

**Pull request merged**